src/libshadertrap/src/executor.cc

```diff
@@ -1201,41 +1203,171 @@ bool Executor::CheckEqualBuffers(CommandAssertEqual* assert_equal) {
     return false;
   }
 
-  uint8_t* mapped_buffer[2]{nullptr, nullptr};
-  for (auto index : {0, 1}) {
-    GL_SAFECALL(&assert_equal->GetStartToken(), glBindBuffer, GL_ARRAY_BUFFER,
-                buffers[index]);
-    mapped_buffer[index] =
-        static_cast<uint8_t*>(gl_functions_->glMapBufferRange_(
-            GL_ARRAY_BUFFER, 0, static_cast<GLsizeiptr>(buffer_size[index]),
-            GL_MAP_READ_BIT));
-    if (mapped_buffer[index] == nullptr) {
-      // TODO(afd): If index == 1 should we unmap buffers[0] before
-      //  returning? Or are we OK so long as we eventually destroy that buffer?
-      GL_CHECKERR(&assert_equal->GetStartToken(), "glMapBufferRange");
-      return false;
-    }
+  bool result = true;
+  size_t start_index = 0;
+  std::vector<CommandAssertEqual::FormatEntry>& format_entries = assert_equal->GetFormatEntries();
+  const Token t = assert_equal->GetStartToken();
+  std::unique_ptr<Token> tmp(std::move(new Token(t)));
+  if(format_entries.size() == 0){
+     format_entries.push_back(
+              {std::move(tmp), CommandAssertEqual::FormatEntry::Kind::kByte, (size_t)buffer_size[0]});
```

Comment:
Use `static_cast<size_t>(buffer_size[0])` instead of a C-style cast.

Reply:
Done